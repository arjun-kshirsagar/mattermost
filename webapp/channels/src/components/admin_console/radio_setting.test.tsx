// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RadioSetting from './radio_setting';

describe('components/admin_console/RadioSetting', () => {
    const baseProps = {
        id: 'string.id',
        label: 'some label',
        values: [
            {text: 'this is engineering', value: 'Engineering'},
            {text: 'this is sales', value: 'Sales'},
            {text: 'this is administration', value: 'Administration'},
        ],
        value: 'Sales',
        onChange: jest.fn(),
        setByEnv: false,
    };

    test('should match snapshot', () => {
        const {container} = render(<RadioSetting {...baseProps}/>);
        expect(container).toMatchSnapshot();
    });

    test('onChange handler is called when radio button is changed', () => {
        render(<RadioSetting {...baseProps}/>);
        
        // Find the first radio button (engineering) by its label text
        const engineeringRadio = screen.getByLabelText('this is engineering');
        
        // Click the radio button
        userEvent.click(engineeringRadio);

        // Verify the onChange handler was called with correct parameters
        expect(baseProps.onChange).toHaveBeenCalledTimes(1);
        expect(baseProps.onChange).toHaveBeenCalledWith('string.id', 'Engineering');
    });
});
