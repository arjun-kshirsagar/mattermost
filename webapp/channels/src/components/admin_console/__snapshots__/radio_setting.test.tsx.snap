// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/RadioSetting should match snapshot 1`] = `
<div>
  <div
    class="form-group"
    data-testid="string.id"
  >
    <label
      class="control-label col-sm-4"
      for="string.id"
    >
      some label
    </label>
    <div
      class="col-sm-8"
    >
      <div
        class="radio"
      >
        <label>
          <input
            name="string.id"
            type="radio"
            value="Engineering"
          />
          this is engineering
        </label>
      </div>
      <div
        class="radio"
      >
        <label>
          <input
            checked=""
            name="string.id"
            type="radio"
            value="Sales"
          />
          this is sales
        </label>
      </div>
      <div
        class="radio"
      >
        <label>
          <input
            name="string.id"
            type="radio"
            value="Administration"
          />
          this is administration
        </label>
      </div>
      <div
        class="help-text"
        data-testid="string.idhelp-text"
      />
    </div>
  </div>
</div>
`;
